src/content/insertable-streams/video-processing/js/main.js

```diff
@@ -165,14 +165,19 @@ function updatePipelineSourceIfSet() {
  */
 function updatePipelineSource() {
   const sourceType = sourceSelector.options[sourceSelector.selectedIndex].value;
-  if (!sourceType) {
+  if (!sourceType || !pipeline) {
     initPipeline();
   } else {
     updatePipelineSourceIfSet();
   }
 }
 sourceSelector.oninput = updatePipelineSource;
-sourceVisibleCheckbox.oninput = () => {
+sourceSelector.disabled = false;
```

Comment:
Can you use https://javascript.info/onload-ondomcontentloaded#window-onload instead or are do you also need to wait for devices to show up (I assume onload only waits for all resources to be (down)loaded and not that all requested hardware is loaded/available)?

Reply:
Is https://github.com/webrtc/samples/pull/1386/commits/e8bb1b2fb73e3b63a3fe2805addab04a05f20bb2 along the lines of what you were thinking? It allows changing the scripts to async, which may help page load delay in some cases.

I also made https://github.com/webrtc/samples/pull/1386/commits/c0dcfc989accccac1642cbce58eeaece48a66800 to put the UI code in a class, but it makes the change harder to review and probably doesn't make things simpler.